app/controllers/PreviouslyRegisteredController.scala

```diff
@@ -49,27 +52,61 @@ class PreviouslyRegisteredController @Inject()(
       Ok(view(form, AdministratorOrPractitioner.Practitioner))
   }
 
-  def onSubmitAdministrator: Action[AnyContent] = Action {
+  def onSubmitAdministrator: Action[AnyContent] = Action.async {
     implicit request =>
       form.bindFromRequest().fold(
         (formWithErrors: Form[_]) =>
-          BadRequest(view(formWithErrors, AdministratorOrPractitioner.Administrator)),
+          Future.successful(BadRequest(view(formWithErrors, AdministratorOrPractitioner.Administrator))),
         {
-          case PreviouslyRegisteredButNotLoggedIn => Redirect(appConfig.recoverCredentialsPSAUrl)
-          case _ => Redirect(appConfig.registerSchemeAdministratorUrl)
+          case PreviouslyRegisteredButNotLoggedIn =>
```

Comment:
We repeat this code for PSAs and PSPs, should we parameterise and make a function we can reuse?

Reply:
moved common code into separate method and re-used for PSAs and PSPs